ingress/controllers/gce/firewalls/firewalls.go

```diff
@@ -60,7 +62,21 @@ func (fr *FirewallRules) Sync(nodePorts []int64, nodeNames []string) error {
 		glog.Infof("Creating global l7 firewall rule %v", name)
 		return fr.cloud.CreateFirewall(suffix, "GCE L7 firewall rule", fr.srcRange, nodePorts, nodeNames)
 	}
-	glog.V(3).Infof("Firewall rule already %v exists, verifying for nodeports %v", name, nodePorts)
+
+	requiredPorts := sets.NewString()
+	for _, p := range nodePorts {
+		requiredPorts.Insert(strconv.Itoa(int(p)))
```

Comment:
Looks odd that there is no Itoa for int64s - also interesting that nodeports is []int64 - seems too large for ports.